src/scene/materials/lit-material.js

```diff
@@ -0,0 +1,262 @@
+import { ShaderProcessorOptions } from '../../platform/graphics/shader-processor-options.js';
+import { CUBEPROJ_NONE, FRESNEL_SCHLICK, GAMMA_SRGBHDR, GAMMA_NONE, LIGHTTYPE_DIRECTIONAL, LIGHTTYPE_OMNI, LIGHTTYPE_SPOT, MASK_AFFECT_DYNAMIC, SPECOCC_AO, SHADER_FORWARDHDR, TONEMAP_LINEAR, SHADERDEF_INSTANCING, SHADERDEF_MORPH_NORMAL, SHADERDEF_MORPH_POSITION, SHADERDEF_MORPH_TEXTURE_BASED, SHADERDEF_SCREENSPACE, SHADERDEF_SKIN, SHADERDEF_NOSHADOW, SHADERDEF_TANGENTS, SPECULAR_BLINN, SPRITE_RENDERMODE_SIMPLE } from "../constants.js";
+import { getProgramLibrary } from "../shader-lib/get-program-library.js";
+import { LitOptions } from "./lit-options.js";
+import { collectLights } from "./lit-material-common.js";
+import { Material } from './material.js';
+import { lit } from '../shader-lib/programs/lit.js';
+
+class LitMaterialOptions {
+    // array of booleans indicating which UV channels are used by the material
+    usedUvs;
+
+    // custom shader chunk to be added to the shader
+    shaderChunk;
+
+    // lit options
+    litOptions = new LitOptions();
+}
+
+class LitMaterialOptionsBuilder {
```

Comment:
I'd vote for a separate file for this class

Reply:
Yes agreed, for options and builder. I was going to do that in the followup PR.

Reply:
Done